refs #001: Kellett milestone 1 migrations

dpfitzsi opened the pull request:
# Summary

** Update from #19 to test build process with different branch name.

This pull request adds migrations for home page, landing page, landing page level 2. This includes process plugins to locate the author and process paragraphs. 

Configuration for content types and associated views will be added as a separate pull request.

# Known Issues

There are no known bugs or issues with code at this time.

# Rollout Instructions

1. Rollout code.
2. Run taxonomy migrations.
3. Import homepage, landing_page, and landing_page_level_2 content and translations.
e.g. `drush migrate:import --tag="Landing Page Level 2"`.

ew-andrew-sus commented on docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml:
```diff
@@ -0,0 +1,75 @@
+destination:
+    plugin: entity:node
+    translations: true
+    destination_module: content_translation
```
add bundle here
default_bundle: homepage

dpfitzsi replied:
This is resolved.

ew-andrew-sus commented on docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page_translations.yml:
```diff
@@ -0,0 +1,75 @@
+        migration: yukon_migrate_home_page
+        source: entity_id
+        no_stub: true
+    type:
```
would be better to add type in destination as default_bundle

dpfitzsi replied:
This is resolved.

ew-andrew-sus reviewed (commented):
Do not use custom plugins to migrate paragraph/media/users please
There are already some migrations ready:
Media: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/media
Paragraphs: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/paragraphs

You can reuse or create similar ones.

ew-andrew-sus commented on docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_home_page.yml:
```diff
@@ -0,0 +1,71 @@
+  promote: promote
+  sticky: sticky
+
+  metatags/0/description: field_page_description/0/value
```
Create a separate field as it was in D7 and migrate value into it. Matatags should be configured to use tat field token. So do not migrate anything directly into matatags please

dpfitzsi replied:
This should now be resolved.

dpfitzsi commented:
> Do not use custom plugins to migrate paragraph/media/users please There are already some migrations ready: Media: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/media Paragraphs: https://github.com/ytgov/yukon-ca/tree/master/docroot/modules/custom/yukon_migrate/migrations/paragraphs
> 
> You can reuse or create similar ones.

I have changed the migration over to yml and added a new source plugin for the one media type (icon).

alt-ew commented on docroot/modules/custom/yukon_migrate/src/Plugin/migrate/source/YukonFilemime.php:
```diff
@@ -0,0 +1,33 @@
+ *   source_module = "file"
+ * )
+ */
+class YukonFilemime extends File {
```
We already have a custom source plugin for files - YukonFile. There is no point to create a new one to add this simple logic. Can you please move this source plugin logic to the YukonFile source plugin?

dpfitzsi replied:
This is now resolved.

alt-ew commented on docroot/modules/custom/yukon_migrate/migrations/paragraphs/yukon_migrate_secondary_content.yml:
```diff
@@ -0,0 +1,34 @@
+    field_category_title: field_category_title
+    field_subcategory_links:
+        -   plugin: sub_process
+            source: field_subcategory_links
```
According to data from the database, this field contains links like 'node/$nid'. During the migration process, we don't keep node ids, therefore we need to replace old node URLs with the new correct ones. For this purpose, we have a new process plugin which is called - Links. Could you please re-work this section to make it use this plugin?

dpfitzsi replied:
This is now resolved.

alt-ew reviewed (changes requested):
Looks like this PR contains changes from other branches.
@dpfitzsi could you please merge the master branch to this branch? Maybe it will resolve the problem.

alt-ew commented on docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2.yml:
```diff
@@ -0,0 +1,76 @@
+    field_icon_dark:
+        plugin: migration_lookup
+        source: field_svg_upload/0/fid
+        migration: yukon_migrate_icon
```
Do we need to add all of the migrations that are used in the migration lookup process to the migration dependencies?

dpfitzsi replied:
This has been resolved

alt-ew commented on docroot/modules/custom/yukon_migrate/migrations/nodes/yukon_migrate_landing_page_level_2_translations.yml:
```diff
@@ -0,0 +1,83 @@
+    content_translation_outdated: translate
+migration_dependencies:
+    required:
+        - yukon_migrate_landing_page_level_2
```
Can you please update the migration dependencies list and add here all required and optional migrations?
For example, your migration has the migration lookup in the yukon_migrate_teams migration, which means that this migration should be a dependency.
Could you please go through all your migrations and fix this? :)

dpfitzsi replied:
This has been resolved
